pkg/service/backup/restore_worker_tables.go

```diff
@@ -155,20 +182,98 @@ func (w *tablesWorker) initHosts(ctx context.Context, run *RestoreRun) error {
 	return nil
 }
 
-func (w *tablesWorker) continuePrevRun() {
-	w.continuation = true
+func (w *tablesWorker) stageRepair(ctx context.Context, run *RestoreRun, target RestoreTarget) error {
+	run.Stage = StageRestoreRepair
+	if run.RepairTaskID == uuid.Nil {
+		run.RepairTaskID = uuid.NewTime()
+	}
+	w.insertRun(ctx, run)
+
+	repairProps, err := json.Marshal(map[string]any{
+		"keyspace": target.Keyspace,
+	})
+	if err != nil {
+		return errors.Wrap(err, "parse repair properties")
+	}
+
+	repairTarget, err := w.repairSvc.GetTarget(ctx, run.ClusterID, repairProps)
+	if err != nil {
+		return errors.Wrap(err, "get repair target")
+	}
+
+	// Insert repair task (as disabled and deleted) into DB so that it can be queried for progress
+	st := &scheduler.Task{
+		ClusterID:  run.ClusterID,
+		Type:       scheduler.RepairTask,
+		ID:         run.RepairTaskID,
+		Name:       "post_restore_repair_" + run.TaskID.String(),
+		Deleted:    true,
+		Properties: repairProps,
+	}
+	if dbErr := table.SchedulerTask.InsertQuery(w.managerSession).BindStruct(st).ExecRelease(); dbErr != nil {
+		w.Logger.Error(ctx, "Insert post-restore repair task",
+			"scheduler_task", *st,
+			"error", dbErr,
+		)
+	}
+
+	repairRunID := uuid.NewTime()
+	str := &scheduler.Run{
+		ClusterID: run.ClusterID,
+		Type:      scheduler.RepairTask,
+		TaskID:    run.RepairTaskID,
+		ID:        repairRunID,
+		Status:    scheduler.StatusRunning,
+		StartTime: timeutc.Now(),
+	}
+	if dbErr := table.SchedulerTaskRun.InsertQuery(w.managerSession).BindStruct(str).ExecRelease(); dbErr != nil {
+		w.Logger.Error(ctx, "Insert post-restore repair task run",
+			"scheduler_task_run", *str,
+			"error", dbErr,
+		)
+	}
+
+	if err = w.repairSvc.Repair(ctx, run.ClusterID, run.RepairTaskID, repairRunID, repairTarget); err != nil {
+		str.Status = scheduler.StatusError
+		str.Cause = err.Error()
+	} else {
+		str.Status = scheduler.StatusDone
+	}
+	endTime := timeutc.Now()
+	str.EndTime = &endTime
+
+	if dbErr := table.SchedulerTaskRun.InsertQuery(w.managerSession).BindStruct(str).ExecRelease(); dbErr != nil {
+		w.Logger.Error(ctx, "Update post-restore repair task run",
+			"scheduler_task_run", *str,
+			"error", dbErr,
+		)
+	}
+	if err != nil {
```

Comment:
My understanding is that even if repair failed we need to rollback tombstone table config. Am I wrong here ?

Reply:
It has always been a bad idea to run two overlapping restores, but checking for that hasn't been done for now. This PR does not change that, so maybe this could be converted into a separate issue?

Reply:
Sure,  https://github.com/scylladb/scylla-manager/issues/3445, feel free to change description, I barely know the topic